transformations/chinese_antonym_synonym_substitution/transformation.py

```diff
@@ -0,0 +1,107 @@
+    random.seed(seed)
+
+    perturbed_texts = []
+    word1_list = [chinese_antonym['word1'] for chinese_antonym in chinese_antonym_data]
+    word2_list = [chinese_antonym['word2'] for chinese_antonym in chinese_antonym_data]
+
+    if(config == 'synonym'):
+        smw = Similarword(create_num=max_outputs+1, change_rate=prob)
+        output_list = smw.replace(text)
+
+        # Notice that the first input is the non-perturb version
+        output_list = output_list[1:]
+        perturbed_texts = output_list
+
+    if(config == 'antonym'):
+
+        output = jieba.lcut(text)
+        for _ in itertools.repeat(None, max_outputs):
+            butter_text = ""
+            for perturb_word in output:
+                perturb_antonym = []
+                if(perturb_word in word1_list):
+                    indices = [index for index, word1 in enumerate(word1_list) if word1 == perturb_word]
+                    perturb_antonym = [word2_list[index] for index in indices]
+                elif(perturb_word in word2_list):
+                    indices = [index for index, word2 in enumerate(word2_list) if word2 == perturb_word]
+                    perturb_antonym = [word1_list[index] for index in indices]
+                else:
+                    print("Word to be perturbed is not in the Chinese Antonym database")
+
+                if random.random() <= prob and len(perturb_antonym) != 0:
+                    new_chinese_character = random.choice(perturb_antonym)
+                else:
+                    new_chinese_character = perturb_word
+                butter_text += new_chinese_character
+            perturbed_texts.append(butter_text)
+    return perturbed_texts
+
+def load_chinese_antonym_data():
+    dirname = os.path.dirname(__file__)
+    words_list = []
+    with open(os.path.join(dirname, '反义词库.txt'), mode='r') as file:
+        lines = file.readlines()
+        for line in lines:
+            if(line.rstrip() != ""):
+                dict = {}
+                word_freq = line.split("-")
+                dict["word1"] = word_freq[0]
+                dict["word2"] = word_freq[1].rstrip()
+                words_list.append(dict)
+    return words_list
+
+"""
+Chinese Antonym And Synonym Substitution
+"""
+
+class ChineseAntonymAndSynonymSubtitution(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION
+    ]
+    languages = ["zh"]
+    keywords = ["lexical", "rule-based", "api-based", "written", "highly-meaning-preserving", "meaning-alteration", "high-precision"]
+
+    def __init__(self, seed=0, max_outputs=1, prob=1):
+        super().__init__(seed, max_outputs=max_outputs)
+        self.prob = prob
+        self.seed = seed
+        self.max_outputs = max_outputs
+        self.chinese_antonym_data = load_chinese_antonym_data()
+
+    def generate(self, sentence: str, config : str = 'synonym'):
+        perturbed_texts = chinese_antonym_synonym_substitution(
```

Comment:
Could you refactor this function to operate on words instead of the entire sentence? The loop over the words in the sentence should be in this `generate` function.